engine/autocompleter.py

```diff
@@ -1,11 +1,106 @@
+"""Contains the titular Autocompleter class, which provides autocompletion suggestions."""
+from __future__ import annotations
+from functools import lru_cache
+from re import split as regex_split
 
 from .term_index import TermIndex, Term
 
+
+class _SuggestionTree:
```

Comment:
There are several options for the observability of this file and with such a small codebase it's less clear what is best.

Other options include:
1. Put in its own module
2. Drop the leading underscore
3. Place in the Autocompleter class

The current option is a middle ground and is an attempt to be a flexible choice.